### DISCUSSION TITLE
WASM example to send data to bevy

### DISCUSSION THREAD

**Discussion by carrascomj** [bevyengine/bevy#6871] [Ideas]
~This is my progress so far (not working)~ Working example: https://github.com/carrascomj/bevying_wasm

## Motivation

Probably more for applications than for games, it is important to upload data from the user during runtime. Doing this on native builds is pretty simple with assets or with some sort of async task.

The problem arises when the application targets WASM. It is very attractive for an application to be run in the web (at least as a showcase for the native build) without the need to installing anything. Because of the limitations of file system access in WASM, events like `FileDragAndDrop` (#6822) won't work. Thus, this requires some extra WASM logic that it is a bit daunting if you haven't work with it in the past ~(I have not succeeded at least)~.

## Proposition

Have an example that implements this logic to deserialize some data coming from the browser and makes it available to bevy's runtime. The approach that I am taking is to use a channel where the sender is owned by a callback and the receiver is a ~`NonSend` resource~ `Resource` that wraps an `async_std::channel::Receiver` (implements `Send`) that can be listened by a bevy system. I do not know if this is the right approach.

There are some successful examples that are beyond what I _think_ should be required for an example:

* https://github.com/intendednull/yewdux-plus-bevy

**Comment by carrascomj** [comment]
The example in the repository that I have linked on top is working, with _a lot_ of comments in the code. Still, I am not sure if using the channel in this way is to be recommended (as a general solution for the problem) or other solutions should be considered.

**Comment by joseph-gio** [comment]
Thanks so much for posting this! It's a huge help.
